test/python/circuit/test_library.py

```diff
@@ -22,15 +22,46 @@
 from qiskit import BasicAer, execute
 from qiskit.circuit import QuantumCircuit, QuantumRegister
 from qiskit.circuit.exceptions import CircuitError
-from qiskit.circuit.library import Permutation, XOR, InnerProduct
+from qiskit.circuit.library import Permutation, XOR, InnerProduct, OR, AND
 from qiskit.circuit.library.arithmetic import (LinearPauliRotations, PolynomialPauliRotations,
                                                IntegerComparator, PiecewiseLinearPauliRotations,
                                                WeightedAdder)
 
 
+@ddt
 class TestBooleanLogicLibrary(QiskitTestCase):
     """Test library of boolean logic quantum circuits."""
 
+    def assertBooleanFunctionIsCorrect(self, boolean_circuit, reference):
```

Comment:
this is nice, we should make a note to rewrite the other boolean logic tests using this.

Reply:
I added notes to the other tests